plugin.yml

```
@@ -32,8 +32,8 @@ configuration:
     cli-version:
       type: string
       enum:
-        - 1
-        - 2
+        - "1"
+        - "2"
```

How about allowing both versions?

```suggestion
      oneOf:
      - type: string
        enum: [ "1", "2" ]
      - type: integer
        enum: [ 1, 2 ]
```

---

Hey @toote, thanks for your feedback.  

My personal taste leans towards unambiguous and clear input requirements, so if it were me i'd pick _either_ a string or integer type for this field.  The reason i picked string (even though the version currently is a number – although in future i could imagine "2.1" being a valid version) is because of the symmetry with being able to provide arguments as environment variables (in this case `BUILDKITE_PLUGIN_DOCKER_COMPOSE_CLI_VERSION`) – environment variables only support string types.  So to avoid a weird impedance mismatch there, i picked string.

However, i totally understand that you may have different trade-offs in mind (such as user friendliness or a different philosophy about strict input validation), and it's your project, so i'm more than happy to adapt my patch to what you decide. 🙌

---

I just went and double-checked, and at least it looks like all the values end up cast to strings anyway so at least we wouldn't have to do extra work to support your integer-or-string schema. 👍 

https://github.com/buildkite-plugins/docker-compose-buildkite-plugin/blob/fe3522acb645f4471014f4218fa7d884024f0b29/lib/shared.bash#L37-L42

---

I've been thinking about it, and after all i think your suggestion is a good one.  I know how much of a pain YAML is with strings vs integers and so on, and it probably can't hurt to support both.  I'll make the changes :+1: